chainer/functions/pooling/unpooling_2d.py

```diff
@@ -44,6 +59,12 @@ def forward(self, x):
         if self.outw is None:
             self.outw = conv.get_deconv_outsize(
                 w, self.kw, self.sx, self.pw, cover_all=self.cover_all)
+        if (self.outh % h == 0 and self.outw % w == 0 and
+            self.outh // h == self.kh and self.outw // w == self.kw and
+            self.ph == 0 and self.pw == 0 and
```

Comment:
Is it difficult to support the case when ph or pw is greater than 0?

Reply:
Hmm, if we have non-zero padding, then the result will not have a uniform pattern, so I don't think we can speed up this cause with stride tricks, but would need another solution, so I think this is out of the scope of this PR.

Reply:
Ok, I realized that it is possible to use stride tricks when `pad % 2 == 0`, but I don't think we can use it with odd padding. I'll update the PR.

Reply:
Ok, I added support for even padding.

Reply:
Great, thanks!